### ISSUE TITLE
Style/EmptyMethod should have the expanded as default

### ISSUE THREAD

**Issue by fidalgo** [rubocop-hq/rubocop#5378]
Based on the Ruby Style guide https://github.com/bbatsov/ruby-style-guide#no-single-line-methods
```
# bad
def too_much; something; something_else; end

# okish - notice that the first ; is required
def no_braces_method; body end

# okish - notice that the second ; is optional
def no_braces_method; body; end

# okish - valid syntax, but no ; makes it kind of hard to read
def some_method() body end

# good
def some_method
  body
end
```
I think the Rubocop's default for the `Style/EmptyMethod` configuration should match the recommendation on the Ruby Styleguide, instead of using the compact option.

I've opened this issue to gather feedback and if we have a consensus I'll open a PR with the changes.

**Comment by bbatsov** [comment]
I don't think there's an explicit rule about this in the style guide, though. Maybe first we should add one. :-)

**Comment by garettarrowood** [comment]
Just to be clear, none of those examples are empty methods. This conflicts with `Style/SingleLineMethods`, where all of those `okish` examples would be considered `bad` by our current implementation. `Style/SingleLineMethods` description reads:

```
This cop checks for single-line method definitions that contain a body.
It will accept single-line methods with no body.
```

So this seems like a feature request to add a config option to `Style/SingleLineMethods` to allow one expression in single line methods.

**Comment by bbatsov** [comment]
Why so? I think @fidalgo just copied the examples from the single-line method rule, therefore my remark we don't actually have a rule about what to do with methods with no body. 

> So this seems like a feature request to add a config option to Style/SingleLineMethods to allow one expression in single line methods.

This might be useful in general, but doesn't seem related to this ticket.

**Comment by garettarrowood** [comment]
All remarks here are valid.  It just seems like citing those examples implies that Rubocop should not register an offense for them. At the moment, those will always register no matter what config the user sets.

Agreed that we need a section in the guide about preferring one-line empty methods over two.

**Comment by fidalgo** [comment]
Well first of all I've made a confusion with `Style/SingleLineMethods` and `Style/EmptyMethod`. 
So I've created a sample file:
```
def some_method
  body
end

def no_op; end

def empty_method
end
```

and got this offence:
`sample.rb:7:1: C: Style/EmptyMethod: Put empty method definitions on a single line.`

which is aligned with the guide. 
So I think we should add a section about empty methods on the Style Guide, because at the moment is just a remark of the Single Line Methods section.

**Comment by Drenmi** [comment]
Correct. When I implemented this, I took this part of the style guide:

> One exception to the rule are empty-body methods.
>
> ```ruby
> # good
> def no_op; end
> ```

as an indication that the above is the recommended for empty methods.

**Comment by bbatsov** [comment]
> So I think we should add a section about empty methods on the Style Guide, because at the moment is just a remark of the Single Line Methods section.

Agreed.
